src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,148 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Create a list of tasks.
+     *
+     * @param tasks the task list
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = new ArrayList<>();
+        for (Task task: tasks) {
+            this.tasks.add(task);
+        }
+    }
+
+    /**
+     * Get the size of the task list.
+     *
+     * @return the size of the task list
+     */
+    public int size() {
+        return tasks.size();
+    }
+
+    /**
+     * Get the task at a specific index in the task list.
+     *
+     * @return the task at a specific index in the task list
+     */
+    public Task get(int getIndex) {
+        return tasks.get(getIndex);
+    }
+
+    /**
+     * Delete the task at a specific index from the task list.
+     *
+     * @return the deleted task
+     */
+    public Task delete(int deleteIndex) {
+        return tasks.remove(deleteIndex);
+    }
+
+    /**
+     * Add the task to the task list.
+     *
+     * @param   task the task to be added
+     */
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Try to make a new to-do task from the input.
+     *
+     * @param  taskString   the input which contains the to-do task information
+     * @return  task    the new to-do task
+     */
+    public Todo makeNewTodoTask(String taskString) throws DukeException {
+        String[] todoTokens = taskString.split(" ");
+
+        // If there is a description
+        if (todoTokens.length > 1) {
+            return new Todo(taskString.replaceFirst("^todo ", ""));
+        }
+
+        throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+    }
+
+    /**
+     * Try to make a new event task from the input.
+     *
+     * @param  taskString   the input which contains the event task information
+     * @return  task    the new event task
+     */
+    public Event makeNewEventTask(String taskString) throws DukeException {
+        String[] eventTokens = taskString.split(" /at ");
+
+        // If there is a description and a time/date
+        if (eventTokens.length > 1) {
+            String dateOrTime = eventTokens[1];
+            String[] dateOrTimeTokens = dateOrTime.split(" ");
+
+            // If there is a date and time
+            if (dateOrTimeTokens.length <= 1) {
+                throw new DukeException("☹ OOPS!!! duke.task.Event tasks require a specific time and date.");
```

Comment:
duke.task.Event , i think you meant "Event" only. this can be a result of refactor - renaming of the class.